### PR TITLE
[INLONG-4774][Manager] Support auth and encryption key user config

### LINKED ISSUE #4774: [Feature][Manager] Support auth and encryption key of user config
### Background

In scenarios where the source of data (e.g. dataproxy sdk or agent) is located outside of idc and the data has to be transferred through insecure / public networks, it is better to enable the clients the ability to authenticate with server and encrypt their data. 

### Use case

Insecure network data transmission.

### Interaction
We use AES encryption to secure data transfer. The AES key is generated each time by the client and is encrypted and decrypted by each client's own RSA key pairs. The RSA key pairs are retrieved from manager when the clients are properly authenticated.

image

### Progress
The function may be implemented with several PRs. 

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
- Fix #4774

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -90,6 +92,9 @@ public boolean create(UserInfo userInfo) {
         entity.setDueDate(getOverDueDate(userInfo.getValidDays()));
         entity.setCreateBy(LoginUserUtils.getLoginUserDetail().getUserName());
         entity.setName(username);
+        entity.setSecretKey(userInfo.getSecretKey());
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:95]: Maybe we should generate the secret key, public key, and private key during saving user info.

Comment (woofyzhao) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:95]: only those who transfer data through public network needs it.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -92,9 +96,19 @@ public boolean create(UserInfo userInfo) {
...
+        try {
+            Map<String, String> keyPairs = RSAUtils.generateRSAKeyPairs();
+            String publicKey = keyPairs.get(RSAUtils.PUBLIC_KEY);
+            String privateKey = keyPairs.get(RSAUtils.PRIVATE_KEY);
+            entity.setPublicKey(publicKey);
+            entity.setPrivateKey(privateKey);
+        } catch (Exception e) {
+            String errMsg = String.format("generate rsa key error: {}", e);
+            log.error(errMsg);
+            throw new BusinessException(errMsg);
```
Comment (gong) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:110]: I suggest print full exception stack.

**Comment by gong** [comment]
Why use RSA encrypt data. RSA mainly is used for key exchange.

**Comment by woofyzhao** [comment]
> Why use RSA encrypt data? RSA mainly is used for key exchange.
It's used to encrypt the generated aes key on public net data transfer scenario.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -105,7 +105,7 @@ public boolean create(UserInfo userInfo) {
             entity.setPublicKey(publicKey);
             entity.setPrivateKey(privateKey);
         } catch (Exception e) {
-            String errMsg = String.format("generate rsa key error: {}", e);
+            String errMsg = String.format("generate rsa key error: %s", e.getMessage());
             log.error(errMsg);
```
Comment (gong) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:109]: Maybe should be `log.error("generate rsa key error ", e)`.

**Review by gong — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java [hunk 4]
```diff
@@ -90,6 +97,19 @@ public boolean create(UserInfo userInfo) {
         entity.setDueDate(getOverDueDate(userInfo.getValidDays()));
         entity.setCreateBy(LoginUserUtils.getLoginUserDetail().getUserName());
         entity.setName(username);
+        entity.setSecretKey(RandomStringUtils.randomAlphanumeric(8));
+        try {
+            Map<String, String> keyPairs = RSAUtils.generateRSAKeyPairs();
+            String publicKey = keyPairs.get(RSAUtils.PUBLIC_KEY);
+            String privateKey = keyPairs.get(RSAUtils.PRIVATE_KEY);
+            entity.setPublicKey(publicKey);
+            entity.setPrivateKey(privateKey);
```
Comment (kipshi) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:106]: In my opinion, SecretKey should be encrypted  once again after persisted in db if possible

Comment (woofyzhao) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/impl/UserServiceImpl.java:106]: Agreed. working on it

**Review by kipshi — APPROVED** [review]

**Review by leosanqing — APPROVED** [review]

**Review by ciscozhou — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by healchow — APPROVED** [review]

**Pull request merged**